src/app/server-cluster/testing/EmptyProvider.h

```diff
@@ -40,6 +40,9 @@ class EmptyProvider : public app::DataModel::Provider
     CHIP_ERROR DeviceTypes(EndpointId endpointId, ReadOnlyBufferBuilder<app::DataModel::DeviceTypeEntry> & builder) override;
     CHIP_ERROR ClientClusters(EndpointId endpointId, ReadOnlyBufferBuilder<ClusterId> & builder) override;
     CHIP_ERROR ServerClusters(EndpointId endpointId, ReadOnlyBufferBuilder<app::DataModel::ServerClusterEntry> & builder) override;
+#if CHIP_CONFIG_USE_ENDPOINT_UNIQUE_ID
+    CHIP_ERROR EndpointUniqueID(EndpointId endpointId, ReadOnlyBufferBuilder<MutableCharSpan> & builder) override;
```

Comment:
```suggestion
    CHIP_ERROR EndpointUniqueID(EndpointId endpointId, MutableCharSpan &out_value) override;
```

seems the correct thing to do and let the caller pass in the buffer to store the value (because we have space for 32 bytes). Otherwise return a const char *, but definitely do not return an array of mutable character spans.